### PR TITLE
Add button analytics

### PR DESCRIPTION
**Pull request by vradopp**
### Reason for changes

Log payment button usage

### Summary of changes

- Added `button_type` which is an existing tag to track the type of button used
- Added events for `payment-button:tapped` and `payment-button:initialized` 

### Checklist

~~- [ ] Added a changelog entry~~

- vradopp

### CODE REVIEW

File: Sources/PaymentButtons/PaymentButton.swift [hunk 1]
```diff
@@ -20,6 +24,12 @@ public class PaymentButton: UIButton {
     static let bundle = Bundle(for: PaymentButton.self)
     #endif
 
+    // Use an empty config and default to live environment for button analytics
+    private let analyticsService = AnalyticsService(
+        coreConfig: .init(clientID: "", environment: .live),
+        orderID: ""
```
Comment (scannillo) [inline Sources/PaymentButtons/PaymentButton.swift]: Nit - is it more clear to pass `"N/A"` instead of `""`?

**Comment by scannillo** [comment]
Changes look good! Another nit - but I think we do want a CHANGELOG entry here, it helps future us track which version we added these analytics in

File: Sources/PaymentButtons/PaymentButton.swift [hunk 2]
```diff
@@ -36,7 +46,9 @@ public class PaymentButton: UIButton {
         self.size = size
         self.insets = insets
         self.label = label
+        self.analyticsService.sendEvent("paypal-button:initialized", buttonType: fundingSource.rawValue)
```
Comment (jaxdesmarais) [inline Sources/PaymentButtons/PaymentButton.swift]: General question: do we want to send a specific event for each `PaymentButton` subclass - so credit, pay later, paypal. Or are we sending the same event name for all buttons?

**Review by scannillo — APPROVED** [review]

File: Sources/PaymentButtons/PaymentButton.swift (hunk 2 above)
Comment (vradopp) [reply Sources/PaymentButtons/PaymentButton.swift]: Its the same event name for all buttons but we also pass the `buttonType` which would be one of `PayPal` `Pay Later` or `Credit`

Comment (jaxdesmarais) [reply Sources/PaymentButtons/PaymentButton.swift]: 🤦 it certainly helps if I could read, sorry just totally missed the second half of this event. Take it or leave it, but is it worth considering `payment-button` as the naming here? Wondering incase we add something like a Venmo button in the future would we reused this class.

Comment (scannillo) [reply Sources/PaymentButtons/PaymentButton.swift]: I like that idea!

Comment (jaxdesmarais) [reply Sources/PaymentButtons/PaymentButton.swift]: @vradopp what do you think about updating the events in this PR to "payment-button" vs "paypal-button"?

Comment (vradopp) [reply Sources/PaymentButtons/PaymentButton.swift]: Sounds good to me! Just updated

**Review by stechiu — APPROVED** [review]
LGTM!

**Review by jaxdesmarais — APPROVED** [review]
🚀 thanks for making that update!

**Comment by saperi22** [comment]
@vradopp 
> Added button_type which is an existing tag to track the type of button used

do you mind pointing me where to look for existing tags? Not sure how to verify if `button_type` is an existing tag.

**Comment by jaxdesmarais** [comment]
> do you mind pointing me where to look for existing tags? Not sure how to verify if `button_type` is an existing tag.

I can help answer this! If you go to `go/fptiexplorer` and click on `tags` on the lefthand side you can search for approved FPTI tags. Generally we want to reuse already approved tags vs creating new tags.

**Pull request merged**